sql/distsql: TestAggregatorAgainstProcessor failed

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2176391&tab=buildLog) on [master@fad5ca5624213f0662f19229b33cb598901ac250](https://github.com/cockroachdb/cockroach/commits/fad5ca5624213f0662f19229b33cb598901ac250):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 36 [running]:
testing.tRunner.func1(0xc000244500)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x42c8e20, 0xc000532680)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x22f
panic(0x42c8e20, 0xc000532680)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x5610800, 0xc000b15440, 0xc000b15440)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0xd8
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc000b1bbc0, 0x55f2f80, 0xc000072120, 0x573ea00, 0xc000b15440, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000c83000, 0xc000b20240, 0x6, 0x6, 0x0, 0x0, 0x0, 0xc000b06be0, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x3e7
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc000c83000, 0xc000b20240, 0x6, 0x6, 0xc000b20240, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0xbe
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc000c83000, 0x3, 0xc000930958, 0x1d5823d, 0xc000119700, 0xc000930958)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x222
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc000c83000, 0x4bf4928, 0xc000119700, 0x55f2f80, 0xc000072120)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x14c
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc000b1b480, 0x1, 0x1, 0xc000b1b4a0, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0xfbf
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000244500)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x1268
testing.tRunner(0xc000244500, 0x4bf4a28)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.43s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2176472&tab=buildLog) on [master@e4e77e2696ed5ff6337ee4393328e5f90022a5f6](https://github.com/cockroachdb/cockroach/commits/e4e77e2696ed5ff6337ee4393328e5f90022a5f6):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 82 [running]:
testing.tRunner.func1(0xc00026ea00)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x688e180, 0xc000858930)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x2ae
panic(0x688e180, 0xc000858930)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x89c1740, 0xc00060bbf0, 0xc00060bbf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0x141
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc0009b23a0, 0x89a3ec0, 0xc000116010, 0x8af8960, 0xc00060bbf0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x9f
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0009e8000, 0xc000812480, 0x8, 0x8, 0x0, 0x0, 0x0, 0xc0009b04b0, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x5ba
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0009e8000, 0xc000812480, 0x8, 0x8, 0xc000812480, 0x8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0x139
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0009e8000, 0xc11fc0, 0xc0009f6578, 0xc000072480, 0xc000148ff8, 0xc00046af00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x168
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0009e8000, 0x71bc808, 0xc000148c00, 0x89a3ec0, 0xc000116010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x289
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc00098f220, 0x1, 0x1, 0xc00098f240, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0x16f1
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc00026ea00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x188d
testing.tRunner(0xc00026ea00, 0x71bc908)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.04s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2176533&tab=buildLog) on [master@fd9632917c4351505d5378753247cd766180a3b0](https://github.com/cockroachdb/cockroach/commits/fd9632917c4351505d5378753247cd766180a3b0):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 37 [running]:
testing.tRunner.func1(0xc000256a00)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x688e1c0, 0xc000e28f50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x2ae
panic(0x688e1c0, 0xc000e28f50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x89c1780, 0xc000bfdec0, 0xc000bfdec0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0x141
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc000c20940, 0x89a3f00, 0xc000074130, 0x8af89a0, 0xc000bfdec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x9f
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000c4e000, 0xc00000c780, 0xa, 0xa, 0x0, 0x0, 0x0, 0xc000ad8550, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x5ba
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc000c4e000, 0xc00000c780, 0xa, 0xa, 0xc00000c780, 0xa)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0x139
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc000c4e000, 0xc00079f8e8, 0xc000c5a578, 0xc000072900, 0xc000134ff8, 0xc000513500)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x168
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc000c4e000, 0x71bc848, 0xc000134c00, 0x89a3f00, 0xc000074130)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x289
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc000c01e80, 0x1, 0x1, 0xc000c01ea0, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0x16f1
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000256a00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x188d
testing.tRunner(0xc000256a00, 0x71bc948)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.66s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2176911&tab=buildLog) on [master@457f1af5d19e0bcef9b5ea9e9125b3332717a86c](https://github.com/cockroachdb/cockroach/commits/457f1af5d19e0bcef9b5ea9e9125b3332717a86c):

Fatal error:
```
panic: expected *DString, found *tree.DTuple [recovered]
	panic: expected *DString, found *tree.DTuple [recovered]
	panic: expected *DString, found *tree.DTuple

```

Stack:
```
goroutine 51 [running]:
testing.tRunner.func1(0xc00049c300)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x42c8f60, 0xc00107bc90)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x22f
panic(0x42c8f60, 0xc00107bc90)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDString(0x5610ae0, 0xc001095590, 0xc001095590, 0x5610ae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:1123 +0xe2
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*concatAggregate).Add(0xc00106cba0, 0x55f3260, 0xc000104010, 0x573e9c0, 0xc00107b690, 0xc00107bc60, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1054 +0x27e
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0010c2c00, 0xc0010a2480, 0x8, 0x8, 0x0, 0x0, 0x0, 0xc001065130, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x3e7
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0010c2c00, 0xc0010a2480, 0x8, 0x8, 0x1, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0xbe
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0010c2c00, 0x0, 0xc000752958, 0x54fb1b8, 0x0, 0xc000752958)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x222
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0010c2c00, 0x4bf4a70, 0xc00013ac00, 0x55f3260, 0xc000104010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x14c
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc0010a1be0, 0x1, 0x1, 0xc0010a1c00, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0xfbf
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc00049c300)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x1268
testing.tRunner(0xc00049c300, 0x4bf4b70)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.54s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2177561&tab=buildLog) on [master@457f1af5d19e0bcef9b5ea9e9125b3332717a86c](https://github.com/cockroachdb/cockroach/commits/457f1af5d19e0bcef9b5ea9e9125b3332717a86c):

Fatal error:
```
panic: expected *DString, found *tree.DTuple [recovered]
	panic: expected *DString, found *tree.DTuple [recovered]
	panic: expected *DString, found *tree.DTuple

```

Stack:
```
goroutine 8 [running]:
testing.tRunner.func1(0xc000bf6400)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x42c8f60, 0xc00034a6a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x22f
panic(0x42c8f60, 0xc00034a6a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDString(0x5610ae0, 0xc0010de5a0, 0xc0010de5a0, 0x5610ae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:1123 +0xe2
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*concatAggregate).Add(0xc0000cfb00, 0x55f3260, 0xc000110010, 0x573ece0, 0xc0010de5a0, 0xc00034a670, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1067 +0x20c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc000d59c00, 0xc0007f82c0, 0x7, 0x7, 0xc000a93c58, 0x1, 0x8, 0xc000fc0cd0, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x3e7
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow(0xc000d59c00, 0xc0007f82c0, 0x7, 0x7, 0xc0007f82c0, 0x7)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:855 +0x1c1
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows(0xc000d59c00, 0x1, 0xc00137a958, 0x1d5837d, 0xc000143180, 0xc00137a958)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:468 +0x234
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next(0xc000d59c00, 0x4bf4a70, 0xc000143180, 0x55f3260, 0xc000110010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:708 +0x14c
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x1, 0x0, 0x0, 0x0, 0xc000a7fca0, 0x1, 0x1, 0xc000a7fcc0, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0xfbf
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000bf6400)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x1268
testing.tRunner(0xc000bf6400, 0x4bf4b70)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.40s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2177562&tab=buildLog) on [master@457f1af5d19e0bcef9b5ea9e9125b3332717a86c](https://github.com/cockroachdb/cockroach/commits/457f1af5d19e0bcef9b5ea9e9125b3332717a86c):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 54 [running]:
testing.tRunner.func1(0xc000278100)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x5f91b00, 0xc0006ac700)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x2ae
panic(0x5f91b00, 0xc0006ac700)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x80c50c0, 0xc00119ee10, 0xc00119ee10)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0x141
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc00119cae0, 0x80a7840, 0xc000120010, 0x81fc2e0, 0xc00119ee10, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x9f
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0011b4c00, 0xc0011a4000, 0x6, 0x6, 0x0, 0x0, 0x0, 0xc0011aa5a0, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x5ba
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0011b4c00, 0xc0011a4000, 0x6, 0x6, 0xc0011a4000, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0x139
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0011b4c00, 0x6b1890, 0xc000a50578, 0xc0008c8d80, 0xc000152ff8, 0xc0004bec00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x168
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0011b4c00, 0x68c0188, 0xc000152c00, 0x80a7840, 0xc000120010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x289
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc00119c260, 0x1, 0x1, 0xc00119c280, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0x16f1
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc000278100)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x188d
testing.tRunner(0xc000278100, 0x68c0288)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.11s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2178044&tab=buildLog) on [master@291c20617fa18a7057b12185f4c5ebe21ec918db](https://github.com/cockroachdb/cockroach/commits/291c20617fa18a7057b12185f4c5ebe21ec918db):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 28 [running]:
testing.tRunner.func1(0xc0004ac900)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x688e2a0, 0xc000665760)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x2ae
panic(0x688e2a0, 0xc000665760)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x89c1880, 0xc000ac10e0, 0xc000ac10e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0x141
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc000ad3680, 0x89a4000, 0xc00011e010, 0x8af8aa0, 0xc000ac10e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x9f
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0002b0800, 0xc000ac6000, 0x9, 0x9, 0x0, 0x0, 0x0, 0xc000ae2690, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x5ba
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0002b0800, 0xc000ac6000, 0x9, 0x9, 0xc000ac6000, 0x9)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0x139
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0002b0800, 0xc11fc0, 0xc000b12578, 0xc000118b40, 0xc000152ff8, 0xc0000e6c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x168
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0002b0800, 0x71bc928, 0xc000152c00, 0x89a4000, 0xc00011e010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x289
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc000ad2ee0, 0x1, 0x1, 0xc000ad2f00, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0x16f1
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc0004ac900)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x188d
testing.tRunner(0xc0004ac900, 0x71bca28)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.07s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2178218&tab=buildLog) on [master@1298e1f8e95604c4f93fdda2be94f626577f7bea](https://github.com/cockroachdb/cockroach/commits/1298e1f8e95604c4f93fdda2be94f626577f7bea):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 16 [running]:
testing.tRunner.func1(0xc00026ca00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x42c9380, 0xc000a3eb10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x22f
panic(0x42c9380, 0xc000a3eb10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x5610f20, 0xc000a541e0, 0xc000a541e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0xd8
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stMakeLineAgg).Add(0xc000a41240, 0x55f36a0, 0xc00010a010, 0x573f120, 0xc000a541e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:580 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0002a9400, 0xc000a534a0, 0xa, 0xa, 0x0, 0x0, 0x0, 0xc0008b79a0, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:812 +0x3e7
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0002a9400, 0xc000a534a0, 0xa, 0xa, 0xc000a534a0, 0xa)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:873 +0xbe
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0002a9400, 0xc0015028e8, 0xc001502958, 0x1d5838d, 0xc00013c000, 0xc001502958)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x222
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0002a9400, 0x4bf4e90, 0xc00013c000, 0x55f36a0, 0xc00010a010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:730 +0x14c
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc000a408e0, 0x1, 0x1, 0xc000a40900, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0xfbf
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc00026ca00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:242 +0x1268
testing.tRunner(0xc00026ca00, 0x4bf4f90)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.26s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/distsql/columnar_operators_test.go
+++ b/pkg/sql/distsql/columnar_operators_test.go
@@ -162,18 +162,22 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 						for j := range aggFnInputTypes {
 							aggFnInputTypes[j] = sqlbase.RandType(rng)
 						}
-						// There is a special case for string_agg when at least
-						// one of the arguments is an empty tuple. Such case
-						// passes GetAggregateInfo check below, but it is
-						// actually invalid, and during normal execution it is
-						// caught during type-checking. However, we don't want
-						// to do fully-fledged type checking, so we hard-code
-						// an exception here.
+						// There is a special case for concat_agg, string_agg,
+						// and st_makeline when at least one argument is a
+						// tuple. Such cases pass GetAggregateInfo check below,
+						// but they are actually invalid, and during normal
+						// execution it is caught during type-checking.
+						// However, we don't want to do fully-fledged type
+						// checking, so we hard-code an exception here.
 						invalid := false
-						if aggFn == execinfrapb.AggregatorSpec_STRING_AGG {
+						switch aggFn {
+						case execinfrapb.AggregatorSpec_CONCAT_AGG,
+							execinfrapb.AggregatorSpec_STRING_AGG,
+							execinfrapb.AggregatorSpec_ST_MAKELINE:
 							for _, typ := range aggFnInputTypes {
-								if typ.Family() == types.TupleFamily && len(typ.TupleContents()) == 0 {
+								if typ.Family() == types.TupleFamily {
 									invalid = true
+									break
 								}
 							}
 						}

--- a/pkg/sql/rowexec/aggregator.go
+++ b/pkg/sql/rowexec/aggregator.go
@@ -555,9 +555,12 @@ func (ag *orderedAggregator) accumulateRows() (
 	return aggEmittingRows, nil, nil
 }
 
+// getAggResults returns the new aggregatorState and the results from the
+// bucket. The bucket is closed.
 func (ag *aggregatorBase) getAggResults(
 	bucket aggregateFuncs,
 ) (aggregatorState, sqlbase.EncDatumRow, *execinfrapb.ProducerMetadata) {
+	defer bucket.close(ag.Ctx)
 	for i, b := range bucket {
 		result, err := b.Result()
 		if err != nil {
@@ -570,7 +573,6 @@ func (ag *aggregatorBase) getAggResults(
 		}
 		ag.row[i] = sqlbase.DatumToEncDatum(ag.outputTypes[i], result)
 	}
-	bucket.close(ag.Ctx)
 
 	if outRow := ag.ProcessRowHelper(ag.row); outRow != nil {
 		return aggEmittingRows, outRow, nil
@@ -640,7 +642,7 @@ func (ag *hashAggregator) emitRow() (
 	// NOTE: accounting for the memory under aggregate builtins in the bucket
 	// is updated in getAggResults (the bucket will be closed), however, we
 	// choose to not reduce our estimate of the map's internal footprint
-	// because it is error-prone to estimate the new footprint (we don't
+	// because it is error-prone to estimate the new footprint (we don't know
 	// whether and when Go runtime will release some of the underlying memory).
 	// This behavior is ok, though, since actual usage of buckets will be lower
 	// than what we accounted for - in the worst case, the query might hit a

